tests/integration/cmd_push_test.go

```diff
@@ -53,6 +53,32 @@ var _ = Describe("odo push command tests", func() {
 			helper.HttpWaitFor("http://"+url, "UPDATED!", 30, 1)
 		})
 
+		It("should be able to create a file, push, delete, then push again propagating the deletions", func() {
+			helper.CmdShouldPass("git", "clone", "https://github.com/openshift/ruby-ex", context+"/ruby-ex")
+			helper.CmdShouldPass("odo", "component", "create", "ruby", cmpName, "--project", project, "--context", context+"/ruby-ex", "--app", appName)
+
+			// Create a new file that we plan on deleting later...
+			newFilePath := filepath.Join(context, "ruby-ex", "foobar.txt")
+			if err := helper.CreateFileWithContent(newFilePath, "hello world"); err != nil {
+				fmt.Printf("the foobar.txt file was not created, reason %v", err.Error())
+			}
+
+			helper.CmdShouldPass("odo", "push", "--context", context+"/ruby-ex")
+
+			// Check to see if it's been pushed (foobar.txt)
+			podName := oc.GetRunningPodNameOfComp(cmpName, project)
+			stdOut := oc.ExecListDir(podName, project)
+			Expect(stdOut).To(ContainSubstring(("foobar.txt")))
+
+			// Now we delete the file and push
+			helper.DeleteDir(newFilePath)
+			helper.CmdShouldPass("odo", "push", "--context", context+"/ruby-ex", "-v4")
+
+			// Then check to see if it's truly been deleted
+			stdOut = oc.ExecListDir(podName, project)
```

Comment:
Please a validation for folder deletion too

Reply:
Already done! :)

Reply:
cool :+1: